src/main/java/org/jboss/modules/LocalModuleFinder.java

```diff
@@ -129,41 +129,37 @@ public LocalModuleFinder(boolean supportLayersAndAddOns) {
         return files;
     }
 
-    private static String toPathString(ModuleIdentifier moduleIdentifier) {
-        final StringBuilder builder = new StringBuilder(40);
-        builder.append(moduleIdentifier.getName().replace('.', File.separatorChar));
-        builder.append(File.separatorChar).append(moduleIdentifier.getSlot());
+    private static String toPathString(String moduleName) {
+        return moduleName.replace('.', File.separatorChar) + File.separatorChar + moduleName + File.separatorChar;
```

Comment:
Similar to above i.e. method body should be:

return moduleName.replace('.', File.separatorChar);